softioc/asyncio_dispatcher.py

```diff
@@ -47,6 +48,25 @@ def close(self):
             self.__atexit = None
 
         self.__shutdown()
+    
+    def wait_for_quit(self):
+
+        async_stop_event = asyncio.Event()
+        stop_event = threading.Event()
+
+        def signal_exit():
+            self.loop.call_soon_threadsafe(async_stop_event.set)
+        
+        async def async_signal_exit():
+            await async_stop_event.wait()
+            stop_event.set()
+
+        for sig in ('SIGINT', 'SIGTERM'):
+            signal.signal(getattr(signal, sig), lambda signum, frame: signal_exit())
```

Comment:
Maybe a little cleaner and more direct to simply write
```python
for sig in (signal.SIGINT, signal.SIGTERM):
```

Reply:
You're right, I'll change the code to this: 
```
        for sig in (signal.SIGINT, signal.SIGTERM):
            signal.signal(sig, lambda signum, frame: signal_exit())
```